src/openrct2/scripting/bindings/entity/ScGuest.cpp

```diff
@@ -472,6 +475,69 @@ namespace OpenRCT2::Scripting
         }
     }
 
+    int16_t ScGuest::rideHeadedTo_get() const
+    {
+        auto peep = GetGuest();
+        if (peep != nullptr)
+        {
+            auto rideId = peep->GuestHeadingToRideId;
+            auto ride = GetRideManager()[rideId];
+            if (ride != nullptr)
+            {
+                return (ride->id.ToUnderlying());
+            }
+        }
+        return -1;
+    }
+
+    void ScGuest::rideHeadedTo_set(uint16_t rideId)
+    {
+        ThrowIfGameStateNotMutable();
+        auto guest = GetGuest();
+        if (guest == nullptr)
+            return;
+
+        Ride* ride = GetRide(RideId::FromUnderlying(rideId));
+        if (ride == nullptr)
+            return;
+
+        if (guest->x == LOCATION_NULL)
+            return;
+
+        /**
+         * Filters out guests who are not in the park or are doing the wrong actions.
+         */
+        PeepState& State = guest->State;
+        if (State != PeepState::Walking && State != PeepState::Sitting && State != PeepState::Watching
+            && State != PeepState::UsingBin && State != PeepState::Queuing && State != PeepState::QueuingFront)
+            return;
+
+        // If the guest is leaving the park, interrupts and sends them to the ride
+        if (guest->PeepFlags & PEEP_FLAGS_LEAVING_PARK)
+        {
+            guest->PeepFlags &= ~PEEP_FLAGS_LEAVING_PARK;
+        }
+
+        if (State == PeepState::Queuing || State == PeepState::QueuingFront)
+        {
+            guest->RemoveFromQueue();
+        }
+
+        // Head to that ride
+        guest->SetState(PeepState::Walking);
+        guest->GuestHeadingToRideId = ride->id;
+        guest->GuestIsLostCountdown = 200;
+        guest->TimeLost = 0;
+        guest->ResetPathfindGoal();
+        guest->WindowInvalidateFlags |= PEEP_INVALIDATE_PEEP_ACTION;
+
+        // Make peep look at their map if they have one
+        if (guest->HasItem(ShopItem::Map))
+        {
+            guest->ReadMap();
+        }
```

Comment:
While a nice touch, maybe it's better if this animation is triggered separately and not through updating this property. This kind of makes it a unexpected side effect of the property.

Reply:
this was copypasted from the original choose ride method, and I do agree. Since it seems like there's an extension on that way that will enable this functionality, i'm happy to remove.